Assets/Scripts/SceneLoader.cs

```diff
@@ -0,0 +1,70 @@
+using Sirenix.OdinInspector;
+using System;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Celeritas
+{
+	public class SceneLoader : MonoBehaviour
+	{
+		[Flags]
+		public enum Scene
+		{
+			Persistent = 1 << 1,
+			Mainmenu = 1 << 2,
+			Main = 1 << 3,
+			RunStart = 1 << 4,
+		}
+
+		[SerializeField]
+		private Scene scene;
+
+		[SerializeField]
+		private LoadSceneMode mode;
+
+		[SerializeField, InfoBox("Load on awake only works in builds, does not occur in editor. Destroys component")]
+		private bool loadOnAwake;
+
+		private void Awake()
+		{
+			if (loadOnAwake)
+			{
+#if !UNITY_EDITOR
+			LoadScene();
+#endif
+				Destroy(this);
+			}
+		}
+
+		public void LoadScene()
+		{
+			string path = "";
+			bool loaded = false;
+
+			if (scene.HasFlag(Scene.Main))
+			{
+				SceneManager.LoadSceneAsync(Constants.MAIN_SCENE_PATH, loaded ? LoadSceneMode.Additive : mode);
+				loaded = true;
+			}
+
+			if (scene.HasFlag(Scene.Persistent))
+			{
+				SceneManager.LoadSceneAsync(Constants.PERSISTENT_SCENE_PATH, loaded ? LoadSceneMode.Additive : mode);
+				loaded = true;
+			}
+
+			if (scene.HasFlag(Scene.Mainmenu))
+			{
+				SceneManager.LoadSceneAsync(Constants.MAINMENU_SCENE_PATH, loaded ? LoadSceneMode.Additive : mode);
+			}
+
+			if (scene.HasFlag(Scene.RunStart))
+			{
+				SceneManager.LoadSceneAsync(Constants.RUNSTART_PATH, loaded ? LoadSceneMode.Additive : mode);
+			}
+
+			if (path != "")
+				SceneManager.LoadScene(path, mode);
```

Comment:
Not a problem, but I'm a bit curious about the significance of this line. is this for scenes that aren't given a flag? Do we want that? Also why isn't it Async like the others?

Reply:
oh that was just left in, it doesn't do anything anymore. I'll remove it